src/gc.c

```diff
@@ -558,7 +558,6 @@ void jl_gc_run_all_finalizers(jl_task_t *ct)
     }
     // unlock here because `run_finalizers` locks this
     JL_UNLOCK_NOGC(&finalizers_lock);
-    gc_n_threads = 0;
```

Comment:
This doesn't seem right to delete

Reply:
This is a local defined 5 lines above, which might be a bug on it's own

Reply:
Oh, that is okay then. I was wondering why we could update this global outside the locked region anyways. Why not the line immediately after it too then?